src/main/java/com/google/api/codegen/transformer/java/JavaGapicSurfaceTestTransformer.java

```diff
@@ -51,6 +54,24 @@
 
   private GapicCodePathMapper pathMapper;
 
+  private class TestCaseNameTable {
+    private HashMap<String, Integer> nameCount;
+
+    public TestCaseNameTable() {
+      nameCount = new HashMap<>();
+    }
+
+    public String getTestName(SurfaceNamer namer, Method method) {
+      String testedMethodName = method.getSimpleName();
+      Integer count = 1;
+      if (nameCount.containsKey(testedMethodName)) {
+        count = nameCount.get(testedMethodName) + 1;
+      }
+      nameCount.put(testedMethodName, count);
+      return namer.getTestCaseName(method, count);
+    }
```

Comment:
My suggestion would be to create a name table that can be used in both spots. Your algorithm is fine, but the SurfaceNamer/Method stuff shouldn't be in the shared code. It's ok to address in a separate PR (but put a TODO here).